HeterogeneousCore/CUDAServices/test/testCUDAService.cpp

```diff
@@ -40,31 +63,26 @@ TEST_CASE("Tests of CUDAService", "[CUDAService]") {
          << ret << ") " << cudaGetErrorString(ret) << ". Running only tests not requiring devices.");
   }
 
-  // Make Service system available as CUDAService depends on ResourceInformationService
-  std::vector<edm::ParameterSet> psets;
-  edm::ServiceToken serviceToken = edm::ServiceRegistry::createSet(psets);
+  std::string config = makeProcess("Test");
+  addCUDAService(config);
+  auto serviceToken = getServiceToken(config);
   edm::ServiceRegistry::Operate operate(serviceToken);
 
-  SECTION("CUDAService enabled") {
-    edm::ParameterSet ps;
-    ps.addUntrackedParameter("enabled", true);
-    SECTION("Enabled only if there are CUDA capable GPUs") {
-      auto cs = makeCUDAService(ps);
-      if (deviceCount <= 0) {
-        REQUIRE(cs.enabled() == false);
-        WARN("CUDAService is disabled as there are no CUDA GPU devices");
-      } else {
-        REQUIRE(cs.enabled() == true);
-        INFO("CUDAService is enabled");
-      }
-    }
-
+  SECTION("Enable the CUDAService only if there are CUDA capable GPUs") {
+    edm::Service<CUDAService> cs;
```

Comment:
@makortel do you think it's better like this, or as
```c++
auto& cs = edm::ServiceRegistry::instance().get<CUDAService>();
```
(and changing all `cs->` back to `cs.`) ?

Reply:
I think the current use is ok. If you want to change back to `cs.`, I'd prefer something along
```
edm::Service<CUDAService> scs;
auto& cs = *scs;
```
so that the implementation details of `edm::Service<T>` are not exposed.